src/arc.rs

```diff
@@ -171,6 +171,42 @@ fn rotate_pt(pt: Vec2, angle: f64) -> Vec2 {
     )
 }
 
+impl ParamCurve for Arc {
+    fn eval(&self, t: f64) -> Point {
+        let angle = self.start_angle + (self.sweep_angle * t);
+        sample_ellipse(self.radii, self.x_rotation, angle).to_point()
+    }
+
+    fn subsegment(&self, range: Range<f64>) -> Self {
+        Self {
+            center: self.center,
+            radii: self.radii,
+            start_angle: self.start_angle + (self.sweep_angle * range.start),
+            sweep_angle: self.sweep_angle - (self.sweep_angle * (1.0 - range.end)),
```

Comment:
This feels wrong, though I haven't done tests to validate it. My intuition says it should be `self.sweep_angle * (range.end - range.start)`. In any case, the code that's there reduces algebraically to `self.sweep_angle * range.end`. It is of course possible I'm missing something.

Reply:
I think what I have is wrong. I'll get back into this and see.

Reply:
I think I've done what you asked here. I think I should add a test.